src/main/java/bio/terra/cli/service/GoogleOauth.java

```diff
@@ -142,98 +143,246 @@ public void browse(String url) {
     }
   }
 
+  /** DRY helper for deleting from DataStore objects parameterized for different types. */
+  private static <T extends Serializable> void deleteFromDataStore(
+      DataStore<T> dataStore, String key) throws IOException {
+    if (!dataStore.containsKey(key)) {
+      logger.debug("Credential for {} not found.", key);
+    }
+    dataStore.delete(key);
+  }
+
   /**
    * Delete the credential associated with the specified userId.
    *
    * @param scopes list of scopes requested of the user
-   * @param clientSecretFile stream to the client secret file
+   * @param clientSecrets client secrets object
    * @param dataStoreDir directory where the local credential store is persisted
    */
   public static void deleteExistingCredential(
-      List<String> scopes, InputStream clientSecretFile, File dataStoreDir)
+      List<String> scopes, GoogleClientSecrets clientSecrets, File dataStoreDir)
       throws IOException, GeneralSecurityException {
-    // load client_secret.json file
-    GoogleClientSecrets clientSecrets =
-        GoogleClientSecrets.load(
-            JSON_FACTORY, new InputStreamReader(clientSecretFile, StandardCharsets.UTF_8));
-
     // get a pointer to the credential datastore
-    GoogleAuthorizationCodeFlow flow = getOAuth2Flow(scopes, clientSecrets, dataStoreDir);
-    DataStore<StoredCredential> dataStore = flow.getCredentialDataStore();
-
-    // check that the specified credential exists
-    if (!dataStore.containsKey(CREDENTIAL_STORE_KEY)) {
-      logger.debug("Credential for {} not found.", CREDENTIAL_STORE_KEY);
-      return;
-    }
-
-    // remove the specified credential
-    dataStore.delete(CREDENTIAL_STORE_KEY);
+    TerraAuthenticationHelper helper =
+        TerraAuthenticationHelper.create(scopes, clientSecrets, dataStoreDir);
+    helper.deleteStoredCredential();
+    helper.deleteStoredIdToken();
   }
 
   /**
    * Get the existing credential for the given user.
    *
    * @param scopes list of scopes requested of the user
-   * @param clientSecretFile stream to the client secret file
+   * @param clientSecrets client secrets object
    * @param dataStoreDir directory where the local credential store is persisted
    * @return credentials object for the user
    */
-  public static UserCredentials getExistingUserCredential(
-      List<String> scopes, InputStream clientSecretFile, File dataStoreDir)
+  public static TerraCredentials getExistingUserCredential(
```

Comment:
Mark this `@Nullable` or, better yet, return `Optional<TerraCredentials>`.

Reply:
Document up here the situations in which the caller needs to refresh the token. I wonder if that couldn't all be done in one call, like `getOrRefreshUserCredentials()`.